nntrainer/models/neuralnet.cpp

```diff
@@ -551,15 +563,25 @@ int NeuralNetwork::train(std::vector<std::string> values) {
   status = allocate(true);
   NN_RETURN_STATUS();
 
-  /** Setup data buffer properties */
-  status = data_buffer->setClassNum(getOutputDimension()[0].width());
-  NN_RETURN_STATUS();
+  auto initiate_data_buffer = [this](std::shared_ptr<DataBuffer> &db) {
+    int status = db->setClassNum(getOutputDimension()[0].width());
+    NN_RETURN_STATUS();
 
-  status = data_buffer->setFeatureSize(getInputDimension()[0]);
-  NN_RETURN_STATUS();
+    status = db->setFeatureSize(getInputDimension()[0]);
```

Comment:
update setFeatureSize() to `setInputShape(std::vector<TensorDim>)`?

Reply:
same here :)